core/services/workflows/v2/config.go

```
@@ -107,10 +109,6 @@ func (c *EngineConfig) Validate() error {
 	if c.Clock == nil {
 		c.Clock = clockwork.NewRealClock()
 	}
-	if c.SecretsFetcher == nil {
-		// TODO: implement
-		c.SecretsFetcher = &unimplementedSecretsFetcher{}
```

Don't we want to return an error if it's missing?

---

We don't require it to be present since we'll set a default in the engine